### PR TITLE
feat(investor-yearn): updated investor-yearn to latest investor-core package  

### PR DESCRIPTION
**Pull request by theoboldfrazier**
BREAKING CHANGE: removes the original api and moves over to investor and opportunity classes

### CODE REVIEW

File: packages/investor-yearn/src/YearnOpportunity.ts [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+        assetReference: this.id
+      }),
+      underlyingPerPosition: bnOrZero(vault.metadata.pricePerShare)
+    }
+    this.feeAsset = {
+      assetId: 'eip155:1/slip44:60'
+    }
+  }
+
+  // TODO(theobold): think through this function to see if we want to call this after yearn
```
Comment (0xdef1cafe) [inline packages/investor-yearn/src/YearnOpportunity.ts:139]: given we're writing brand new code, i'd prefer if we consider if we need this or not now.

given this class member is private, and seemingly unused, can we simply remove it?

Comment (cjthompson) [reply packages/investor-yearn/src/YearnOpportunity.ts:142]: We left it in as a reminder for us to circle back to the question of whether we need this type of function or not, but I'm okay with removing it until we do.

**Review by cjthompson — DISMISSED** [review]

File: packages/investor-yearn/src/yearncli.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      wsUrl: 'ws://api.ethereum.shapeshift.com'
+    }
+  }
+  const adapterManager = new ChainAdapterManager(unchainedUrls)
+  const wallet = await getWallet()
+  const chainAdapter = adapterManager.byChainId(
+    toChainId({ chainNamespace: 'eip155', chainReference: '1' })
+  ) as ChainAdapter<ChainTypes.Ethereum>
+  const yearnInvestor = new YearnInvestor({
+    providerUrl: 'https://daemon.ethereum.shapeshift.com', // 'https://api.ethereum.shapeshift.com',
```
Comment (0xdef1cafe) [inline packages/investor-yearn/src/yearncli.ts:41]:
```suggestion
    providerUrl: 'https://daemon.ethereum.shapeshift.com',
```

File: packages/investor-yearn/src/YearnOpportunity.ts [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+  //   // re-fetch data and update state
+  //   const contract = new this.#internals.web3.eth.Contract(yv2VaultAbi, this.id)
+  //   const pricePerShare = await contract.methods.pricePerShare().call()
+  //   return bnOrZero(pricePerShare)
+  //   const totalSupply = await contract.methods.totalSupply().call()
+  //   return bnOrZero(totalSupply)
+  //   const positionBalance = await contract.methods.balanceOf(userAddress).call()
+  //   return bnOrZero(positionBalance)
+  // }
+
```
Comment (0xdef1cafe) [inline packages/investor-yearn/src/YearnOpportunity.ts:154]:
```suggestion
```

File: packages/investor-yearn/src/yearncli.ts (hunk 2 above)
Comment (theoboldfrazier) [reply packages/investor-yearn/src/yearncli.ts]: Removed comment

File: packages/investor-yearn/src/YearnOpportunity.ts (hunk 3 above)
Comment (theoboldfrazier) [reply packages/investor-yearn/src/YearnOpportunity.ts]: removed

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**